petri/src/vm/mod.rs

```diff
@@ -67,6 +67,8 @@ pub trait PetriVmConfig: Send {
     fn with_uefi_frontpage(self: Box<Self>, enable: bool) -> Box<dyn PetriVmConfig>;
     /// Run the VM with Enable VMBus relay enabled
     fn with_vmbus_redirect(self: Box<Self>, enable: bool) -> Box<dyn PetriVmConfig>;
+    /// Run the VM with SCSI relay through VTL2 enabled
+    fn with_scsi_relay(self: Box<Self>, enable: bool) -> Box<dyn PetriVmConfig>;
```

Comment:
instead of using this approach, leverage the existing `.with_custom_config` and `.with_custom_vtl2_settings` machinery. If those methods are private in the OpenVMM support in Petri, promote them to be a member of the `PetriVm` or `PetriVmConfig` trait, whichever is necessary.